app/components/blacklight/document/bookmark_component.rb

```diff
@@ -21,6 +21,14 @@ def bookmarked?
         helpers.bookmarked? @document
       end
 
+      def bookmark_icon
+        icon = helpers.blacklight_icon("bookmark")
```

Comment:
This is a deprecated method and calling it is going to cause a deprecation warning. https://github.com/projectblacklight/blacklight/blob/e957632975c4e190ed94a812bee74d774a5459af/app/helpers/blacklight/icon_helper_behavior.rb#L12-L20

Reply:
Without this I have to create a component which I haven't figured out a good way to switch between in tests. Do you have any ideas?

Reply:
Can we make it a configuration setting like `config.bookmark.use_icon` and set it to false by default?  That's a pretty good way of having optional features as we use it for other things as well.